src/views/panel.rs

```diff
@@ -47,4 +118,9 @@ impl<V: View> ViewWrapper for Panel<V> {
         let inner_size = size.saturating_sub((2, 2));
         self.view.important_area(inner_size) + (1, 1)
     }
+
+    fn wrap_needs_relayout(&self) -> bool {
+        self.invalidated
+            || self.with_view(|v| v.needs_relayout()).unwrap_or(true)
```

Comment:
In this case you can directly call `self.v.needs_relayout()` I think? The `with_view` interface is to deal with more complex wrappers where it can't always access the inner view (think `RefCell`).